src/tools/autosave.js

```diff
@@ -1,123 +1,65 @@
-import db from "./autosave-client";
+import { getLocalAutosaveState, updateLocalAutosave } from "./local-autosave";
+import {
+  initializeServerAutoSave,
+  scheduleServerAutoSave
+} from "./server-autosave";
 import { connectionModeIsServer } from "./server-tools";
-import { setPreviousAutosave, loadAutosave } from "../actions/actions";
+import { store } from "../store";
```

Comment:
In general I agree, unfortunately in this case we kind of need to do this, as the redux store might be out of date by the time the callback fires. Same with the case you noticed below. You'll note that the main function `updateAutoSave` in this file takes a state parameter, rather than using the store.

That said, it's possible there is a better way of organizing this than what I came up with. Suggestions welcome.

Reply:
Actually, looking at this again, I don't think we need to import store. Will fix.